src/vs/platform/files/common/files.ts

```diff
@@ -570,19 +605,22 @@ export class FileChangesEvent {
 	 * Returns if this event contains updated files.
 	 */
 	gotUpdated(): boolean {
-		return this.hasType(FileChangeType.UPDATED);
+		return !!this.updated;
 	}
 
 	private getOfType(type: FileChangeType): IFileChange[] {
-		return this.changes.filter(change => change.type === type);
-	}
+		const changes: IFileChange[] = [];
+
+		const eventsForType = type === FileChangeType.ADDED ? this.added : type === FileChangeType.UPDATED ? this.updated : this.deleted;
+		if (eventsForType) {
+			eventsForType.forEach(change => changes.push(change));
```

Comment:
@jrieken maybe `TernarySearchTree` could expose a method to get the values as array

Reply:
Yeah, maybe it should be iterable but it isn't so simple when I tried

Reply:
Adopted your change: https://github.com/microsoft/vscode/pull/108096/commits/b7a8cf410545c32aeb93bccda9369f349267228e#diff-fe70f37a11beee27d8bfd68ecfc6822fR616